sdks/python/src/opik/config.py

```diff
@@ -129,6 +129,13 @@ def settings_customise_sources(
     The amount of background threads that submit data to the backend.
     """
 
+    file_upload_background_workers: Optional[int] = 16
+    """
+    The amount of background threads that upload files to the backend.
+
+    If not specified, it will default to the number of CPU cores, but no more than 32 workers.
```

Comment:
I don't think we should be bound to the amount of CPU cores (and probably we can even ignore the amount of cores when creating a pool).
File upload is an IO-operation, so we can easily exceed the number of cores, the bottleneck is the network bandwidth anyway.